# GARFIELD
GARFIELD: Genomic vARiants FIltering by dEep Learning moDels
<br> By Viola Ravasio and Edoardo Giacopuzzi (edoardo.giacopuzzi@unibs.it)
Version 1.0

USAGE: Predict.pl --input input.vcf[.gz] --output output.vcf --platform [illumina/ion]

GARFIELD is implemented as Perl script and requires java to perform predictions. 

The script takes an input vcf (or vcf.gz) file and adds Ptrue values for each variants in INFO field, generating a new file VCF. Prediction scores are marked as [Sample Name]_true=[value] in the output VCF. Platform for prediction [ion/illumina] must be specified by --platform

GARFIELD requires VCF files generated by GATK for Illumina platform or TVC (Torrent Variant Caller) for ION. Multisample VCF can be processed, an independent Ptrue value is added in output file for each sample in the format [Sample Name]_true=[value].
The tool returns a standard VCF output and can thus be easily integrated in already established analysis pipelines.

Predictions are made based on four deep learning models specifically trained on INDELs and SNPs from Illumina and ION platforms. Calcualted Ptrue value ranges from 0 to 1, expressing the proabilit of a true variant.
While canonical GATK and ION hard-filters generate boolean values, GARFIELD calculates a continuous prediction value, allowing tuning of variant filtering depending on the desired accuracy and specificity. 

Ptrue threshold for > 99% TPR (alias sensitivity):
<br><bold>ION</bold> SNPs  0.9198  INDELs  0.031
<br><bold>Illumina</bold> SNPs 0.0354  INDELs  0.0262

Ptrue threshold for max accuracy (achived sensitivity):
<br><bold>ION</bold> SNPs  0.7203 (0.997) INDELs 0.4778 (0.97)
<br><bold>Illumina</bold> SNPs 0.0112 (0.996)  INDELs 0.1249 (0.976)

See [Article] for additional filtering threshold and detailed statistics.

If you found GARFIELD useful please cite this github and our paper:
[ARTICLE]
